### PR TITLE
[FIX] project,sale_timesheet: fix project update right panel values

### PR DESCRIPTION
**Pull request by cedricpri**
Prior to this commit:
- Some of the project update right panel values were wrong (the number of sold and
effective hours were wrong if the service is in units, and eventual down payments are
now removed from the list of SOLs)

task-2703516

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/project/models/project.py [hunk 1]
```diff
@@ -2088,14 +2088,14 @@ def action_dependent_tasks(self):
             'res_model': 'project.task',
             'type': 'ir.actions.act_window',
             'context': {**self._context, 'default_depend_on_ids': [Command.link(self.id)]},
+            'domain': [('depend_on_ids', '=', self.id)],
```
Comment (xavierbol) [inline addons/project/models/project.py:2091]: I don't understand this change 🤔
Perhaps I miss something. 😇

File: addons/sale_timesheet/models/project.py [hunk 2]
```diff
@@ -305,12 +305,12 @@ def get_panel_data(self):
             'profitability_items': self._get_profitability_items(),
         }
 
-    def _get_sale_order_lines(self):
+    def _get_sale_order_lines_not_down_payments(self):
```
Comment (xavierbol) [inline addons/sale_timesheet/models/project.py:308]: In stable, we cannot change the signature of the method, that is we cannot change the return type, the parameters and the method name.
So, we should keep the method name as before. 😇 
```suggestion
    def _get_sale_order_lines(self):
```

File: addons/sale_timesheet/models/project.py [hunk 3]
```diff
@@ -305,12 +305,12 @@ def get_panel_data(self):
...
         sale_orders = self.sale_order_id | self.tasks.sale_order_id
-        return self.env['sale.order.line'].search([('order_id', 'in', sale_orders.ids), ('is_service', '=', True)], order='id asc')
+        return self.env['sale.order.line'].search([('order_id', 'in', sale_orders.ids), ('is_service', '=', True), ('is_downpayment', '=', False)], order='id asc')
 
     def _get_sold_items(self):
-        sols = self._get_sale_order_lines()
+        sols = self._get_sale_order_lines_not_down_payments()
```
Comment (xavierbol) [inline addons/sale_timesheet/models/project.py:313]:
```suggestion
        sols = self._get_sale_order_lines()
```

File: addons/sale_timesheet/models/project_update.py [hunk 4]
```diff
@@ -34,7 +34,7 @@ def _get_services_values(self, project):
             return {}
         services = []
         total_sold, total_effective, total_remaining = 0, 0, 0
-        sols = project._get_sale_order_lines()
+        sols = project._get_sale_order_lines_not_down_payments()
```
Comment (xavierbol) [inline addons/sale_timesheet/models/project_update.py:37]:
```suggestion
        sols = project._get_sale_order_lines()
```

**Review by xavierbol — COMMENTED** [review]
Thanks for the fix, I have made some comments. 🙂

File: addons/project/models/project.py (hunk 1 above)
Comment (cedricpri) [reply addons/project/models/project.py:2091]: This is supposed to fix the following point of the task: project.task: the 'blocking' stat button is missing a domain. Is there something wrong here?

Comment (xavierbol) [reply addons/project/models/project.py:2073]: Oh okay sorry 😇

**Review by xavierbol — APPROVED** [review]
LGTM, thanks ! 🙂

**Review by IT-Ideas — APPROVED** [review]
Thanks for your work 🙂

**Comment by IT-Ideas** [comment]
Hello @cedricpri 
Can you rebase and also limit the commit message to the diff. Here it seems to be the scope of both the enterprise and community PR.
Thanks!

**Comment by IT-Ideas** [comment]
@robodoo r+

**Pull request closed without merging**